src/encode.ts

```diff
@@ -55,18 +56,29 @@ const encoder = new Encoder()
 /**
  * Encode and yield one or more messages
  */
-export async function * encode (source: Source<Message | Message[]>) {
```

Comment:
Simplification - it's only ever `Source<Message>` in the tests, so just make it `Source<Message[]>`.